### PR TITLE
Add OTEL_EXPORTER_JAEGER_TIMEOUT environment variable

### PR DESCRIPTION
**Pull request by trask**
Adds `OTEL_EXPORTER_JAEGER_TIMEOUT` environment variable, similar to `OTEL_EXPORTER_OTLP_TIMEOUT`.

The Java SDK already has this as a programmatic option:

https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/jaeger/src/main/java/io/opentelemetry/exporter/jaeger/JaegerGrpcSpanExporterBuilder.java#L76-L83

and we would like to surface this as an environment variable.

### CODE REVIEW

**Review by jkwatson — APPROVED** [review]

File: specification/sdk-environment-variables.md [hunk 1]
```diff
@@ -91,13 +91,14 @@ See [OpenTelemetry Protocol Exporter Configuration Options](./protocol/exporter.
...
+| Name                            | Description                                                  | Default                                                                                          |
+|---------------------------------|--------------------------------------------------------------|--------------------------------------------------------------------------------------------------|
+| OTEL_EXPORTER_JAEGER_AGENT_HOST | Hostname for the Jaeger agent                                | "localhost"                                                                                      |
+| OTEL_EXPORTER_JAEGER_AGENT_PORT | Port for the Jaeger agent                                    | 6832                                                                                             |
+| OTEL_EXPORTER_JAEGER_ENDPOINT   | HTTP endpoint for Jaeger traces                              | <!-- markdown-link-check-disable --> "http://localhost:14250"<!-- markdown-link-check-enable --> |
+| OTEL_EXPORTER_JAEGER_TIMEOUT    | Max waiting time for the backend to process each spans batch | 10s                                                                                              |
```
Comment (iNikem) [inline specification/sdk-environment-variables.md:99]: It sounds like it limits server time. Is it timeout for the exporter (client) or e.g. collector's receiver (server)?

Comment (jkwatson) [reply specification/sdk-environment-variables.md:99]: probably should be reworded to "Max waiting time for the backend to respond to the client for a request" or something like that.

Comment (reyang) [reply specification/sdk-environment-variables.md:99]: "Maximum time that the Jaeger exporter would wait for each batch export."

Comment (jkwatson) [reply specification/sdk-environment-variables.md:99]: It's probably important to call out that it's the request timeout and not a network connect timeout.

Comment (trask) [reply specification/sdk-environment-variables.md]: Updated, also applied the new wording to `OTEL_EXPORTER_OTLP_TIMEOUT` which is where I copied this wording from to begin with 😁

**Review by carlosalberto — APPROVED** [review]

**Review by jmacd — APPROVED** [review]

**Comment by jmacd** [comment]
Sorry, would you please resolve the merge conflicts?

**Pull request merged**